File: app/src/main/java/com/haishinkit/amf/data/ASArray.java
package com.haishinkit.amf.data;

import java.util.ArrayList;
import java.util.Map;
import java.util.HashMap;

public final class ASArray extends ArrayList<Object> {
    private Map<String, Object> properties = new HashMap<String, Object>();

    public ASArray(int capacity) {
        super(capacity);
    }

    public void put(String k, Object v) {
        properties.put(k, v);
    }

    public String toString() {
        return "{" + super.toString() + "," + properties.toString() + "}";
    }
}

File: haishinkit/src/main/java/com/haishinkit/amf/data/AsArray.kt
package com.haishinkit.amf.data

class AsArray(capacity: Int) : ArrayList<Any>(capacity) {
    private val properties = HashMap<String, Any?>()

    fun put(
        k: String,
        v: Any?,
    ) {
        properties.put(k, v)
    }

    override fun toString(): String {
        return "{" + super.toString() + "," + properties.toString() + "}"
    }
}
